### DISCUSSION TITLE
Call to undefined method Canvas\Models\User::authentications()

### DISCUSSION THREAD

**Discussion by SaschaNe** [austintoddj/canvas#1350] [Troubleshooting] [answered]
Hey, the installation works fine, but as soon as I log in, it shows me the following error.

https://flareapp.io/share/yPaxEEwm#stack

##PHP Version:
PHP 8.2.10 (cli) (built: Sep 2 2023 06:59:22) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.10, Copyright (c) Zend Technologies
with Zend OPcache v8.2.10, Copyright (c), by Zend Technologies

##Composer.json:
```
"require": {
    "php": "^8.1",
    "backpack/crud": "^6.2",
    "doctrine/dbal": "^3.6",
    "geoip2/geoip2": "^2.13",
    "guzzlehttp/guzzle": "^7.2",
    "joelbutcher/socialstream": "^4.4",
    "laravel/framework": "^10.10",
    "laravel/jetstream": "^4.0",
    "laravel/sanctum": "^3.2",
    "laravel/tinker": "^2.8",
    "livewire/livewire": "^3.0",
    "rappasoft/laravel-authentication-log": "^3.0",
    "torann/geoip": "^3.0"
},
```

**Comment by austintoddj** [comment]
Hi @SaschaNe - just spun up a fresh install on the latest version of Canvas and I couldn't reproduce the error. Any more details you can provide?

**Reply by SaschaNe** [reply]
I'm sorry, I don't have more details, I already provided everything. The error, the composer.json, and the Version I'm using. Which more information do you need? :-)

**Reply by austintoddj** [reply, marked as answer]
@SaschaNe I was able to reproduce the error. It's colliding with the `rappasoft/laravel-authentication-log` package. If you remove that from your composer.json file the authentication works.

**Reply by SaschaNe** [reply]
Here is the whole ZIP file: https://droppy24.com/ (will be deleted in a few days)

**Reply by austintoddj** [reply]
No need for the zip file, I was able to repro the issue. I don't know what laravel-authentication-log is doing under the hood.

**Reply by SaschaNe** [reply]
Do we have a solution?

**Reply by austintoddj** [reply]
Their package seems like [it's listening for a User login event](https://github.com/rappasoft/laravel-authentication-log/blob/main/src/Listeners/LoginListener.php#L19). They're also assuming that you've [configured your model](https://rappasoft.com/docs/laravel-authentication-log/v1/start/configuration#content-setting-up-your-model) with their `AuthenticationLoggable` and `Notifiable` traits.

Canvas isn't going to utilize this, so it won't be on the User model. Like your screenshot shows, it's looking for `Canvas\Models\User::authentications()` which is [being called here](https://github.com/rappasoft/laravel-authentication-log/blob/main/src/Listeners/LoginListener.php#L33).

I'd reach out that package maintainer and see if there's a way to _not listen_ for specific models or events, or rule out anything that's not coming from your base Laravel user authentication.

**Comment by austintoddj** [comment]
Closing this out with [Canvas v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0).

Canvas no longer ships a package `User` model for auth. Identity is the **host** user model; Canvas access lives in `canvas_users`. Trait packages that expected `Canvas\Models\User` no longer apply that way.

See the [upgrade guide](https://github.com/austintoddj/canvas/blob/main/.github/UPGRADE.md) for the host-auth contract and `HasCanvasAccess` (optional).
